File: fortran/difference-of-squares/difference_of_squares.f90
module difference_of_squares
  implicit none
contains

  pure integer function sum_of_squares(n)
    integer, intent(in) :: n
    sum_of_squares = n * (n + 1) * (2 * n + 1) / 6
  end function sum_of_squares

  pure integer function square_of_sum(n)
    integer, intent(in) :: n
    square_of_sum = (n * (n + 1) / 2) ** 2
  end function square_of_sum

  pure integer function difference(n)
    integer, intent(in) :: n
    difference = square_of_sum(n) - sum_of_squares(n)
  end function difference

end module difference_of_squares

File: euphoria/difference-of-squares/difference-of-squares.ex
public function squareOfSum(integer limit)
  integer sum = 0
  for n = 1 to limit do
      sum += n
  end for
  return sum * sum
end function

public function sumOfSquares(integer limit)
  integer sum = 0
  for n = 1 to limit do
      sum += n * n
  end for
  return sum
end function

public function differenceOfSquares(integer n)
  return squareOfSum(n) - sumOfSquares(n)
end function
